src/bio/bio_buffer.c

```diff
@@ -992,8 +995,8 @@ bio_rwv(struct bio_io_context *ioctxt, struct bio_sglist *bsgl_in,
 		goto out;
 
 	for (i = 0; i < bsgl->bs_nr; i++) {
-		D_ASSERT(bsgl_in->bs_iovs[i].bi_buf == NULL);
-		D_ASSERT(bsgl_in->bs_iovs[i].bi_data_len != 0);
+		D_ASSERT(bio_iov2raw_buf(&bsgl_in->bs_iovs[i]) == NULL);
+		D_ASSERT(bio_iov2raw_len(&bsgl_in->bs_iovs[i]) != 0);
```

Comment:
use the bio_iov2buf() and bio_iov2len() which includes some sanity check over the extra lens?

Reply:
changed